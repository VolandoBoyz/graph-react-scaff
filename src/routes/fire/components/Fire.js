import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Button from '../../../components/Button';
import fireImg from '../../../assets/Fire_emoji.png';

class Fire extends Component {
  componentDidMount() {
    this.props.fetchFire();
  }
  render() {
    return (
      <section className={`${this.props.className} fire-container`}>
        <section className='image-container'>
          <img src={fireImg} alt='fire emoji' />
        </section>
        <section className='action-buttons'>
          <Button onClick={this.props.incrementFire}>Increment</Button>
          <Button onClick={this.props.decreaseFire}>Decrease</Button>
        </section>
      </section>
    );
  }
}

const FireStyled = styled(Fire)`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  .image-container{
    height: calc(100vh - 64px - 80px);
    display: flex;
    justify-content: center;
    align-items: center;
    img {
      transform: ${props => `scale(${1 + props.fire})`};
      transition: all ease-in-out 0.1s;
    }
  }
  .action-buttons{
    z-index: 1;
    height: 80px;
  }
`;

Fire.propTypes = {
  incrementFire: PropTypes.func.isRequired,
  decreaseFire: PropTypes.func.isRequired,
  fetchFire: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired,
};

export default FireStyled;
