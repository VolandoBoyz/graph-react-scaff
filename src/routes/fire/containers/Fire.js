import { connect } from 'react-redux';
import { decreaseFire, incrementFire, fetchFire } from '../../../modules/fire';
import Fire from '../components/Fire';

const mapDispatchToprops = {
  decreaseFire,
  incrementFire,
  fetchFire,
};

const mapStateToProps = state => ({
  fire: state.fire.fireSize,
});

export default connect(mapStateToProps, mapDispatchToprops)(Fire);
