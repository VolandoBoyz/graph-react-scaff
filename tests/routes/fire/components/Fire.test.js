import React from 'react';
import renderer from 'react-test-renderer';
import Fire from '../../../../src/routes/fire/components/Fire';

test('Fire', () => {
  let fire = 0;
  const component = renderer.create(
    <Fire
      fire={fire}
      incrementFire={() => { fire += 1; }}
      decreaseFire={() => { fire -= 1; }}
      fetchFire={() => 'ðŸ”¥'}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
