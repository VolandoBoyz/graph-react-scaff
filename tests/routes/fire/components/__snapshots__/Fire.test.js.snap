// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fire 1`] = `
<section
  className="gbQwjE fire-container"
>
  <section
    className="image-container"
  >
    <img
      alt="fire emoji"
      src="test-file-stub"
    />
  </section>
  <section
    className="action-buttons"
  >
    <button
      className="hDNYrD"
      onClick={[Function]}
    >
      Increment
    </button>
    <button
      className="hDNYrD"
      onClick={[Function]}
    >
      Decrease
    </button>
  </section>
</section>
`;
